client/components/LearnerViewDirective.js

```diff
@@ -410,11 +469,21 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        $scope.submitCode = function(code) {
+        var callSolutionHandlerService = function() {
+          var code = this.code;
           SolutionHandlerService.processSolutionAsync(
-            prompts[currentPromptIndex], code,
-            question.getAuxiliaryCode(language), language
-          ).then(setFeedback);
+              prompts[currentPromptIndex], code,
+              question.getAuxiliaryCode(language), language
+              ).then(setFeedback);
+        };
+
+        $scope.submitCode = function(code) {
+          $scope.hideDotDiv = false;
+          $timeout(
+              function(){feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17},
+              0);
+          var serviceCall = callSolutionHandlerService.bind({'code': code});
+          $timeout(serviceCall, 1000);
```

Comment:
Ah yes... as it's written, a minimum 15-20ms (at least on my laptop) is needed for the scrolling to work (and 0ms for the scrollTop change). I don't like having any timeouts > 0 either, but.. any suggestions?